src/main/kotlin/com/team4099/robot2020/subsystems/Feeder.kt

```diff
@@ -0,0 +1,81 @@
+package com.team4099.robot2020.subsystems
+
+import com.ctre.phoenix.motorcontrol.ControlMode
+import com.ctre.phoenix.motorcontrol.can.TalonSRX
+import com.team4099.lib.motorcontroller.CTREMotorControllerFactory
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants
+
+object Feeder : Subsystem {
+    private val inMasterTalon: TalonSRX
+    private val inSlaveVictor = CTREMotorControllerFactory.createPermanentSlaveVictor(
+            Constants.Feeder.FEEDER_IN_MASTER_TALON_ID,
+            Constants.Feeder.FEEDER_IN_SLAVE_VICTOR_ID
+    )
+
+    private val outTalon: TalonSRX
+
+    var feederState = FeederState.HOLD
+    private var feederInPower = 0.0
+    private var feederOutPower = 0.0
+
+    init {
+        inMasterTalon = CTREMotorControllerFactory.createDefaultTalon(Constants.Feeder.FEEDER_IN_MASTER_TALON_ID)
+
+        outTalon = CTREMotorControllerFactory.createDefaultTalon(Constants.Feeder.FEEDER_OUT_TALON_ID)
+
+        inMasterTalon.inverted = false
+        inSlaveVictor.inverted = true
+
+        outTalon.inverted = true
+    }
+
+    enum class FeederState {
+        HOLD, INTAKE, SHOOT
+    }
+
+    private fun setFeederInPower(power: Double) {
+        inMasterTalon.set(ControlMode.PercentOutput, power)
+    }
+
+    private fun setFeederOutPower(power: Double) {
+        outTalon.set(ControlMode.PercentOutput, power)
+    }
+
+    override fun outputTelemetry() { }
+
+    override fun checkSystem() { }
+
+    override fun registerLogging() { }
+
+    override fun zeroSensors() { }
+
+    override fun onStart(timestamp: Double) {
+        feederState = FeederState.HOLD
+    }
+
+    override fun onLoop(timestamp: Double, dT: Double) {
+        synchronized(this@Feeder) {
```

Comment:
can do `@Synchronized` because this is actually feeder here (this is why subsystem implementing loop makes so much sense)